apt/method.go

```diff
@@ -114,6 +114,12 @@ func (m *AptMethod) initClient() error {
 		ts = creds.TokenSource
 	case m.config.serviceAccountEmail != "":
 		ts = google.ComputeTokenSource(m.config.serviceAccountEmail)
+	case m.config.accessTokenENV != "":
+		accessToken := os.Getenv(m.config.accessTokenENV)
+		if accessToken == "" {
+			return fmt.Errorf("Failed to obtain token from environment variable: %s", m.config.accessTokenENV)
+		}
+		ts = oauth2.StaticTokenSource(&oauth2.Token{AccessToken: accessToken})
```

Comment:
Is it guaranteed that this access token will remain fresh for the duration of the session? Note how the other auth methods use `google.ComputeTokenSource` and `google.CredentialsFromJSON`, which both include logic for refreshing the token: https://cs.opensource.google/go/x/oauth2/+/refs/tags/v0.13.0:google/google.go;l=253

I suspect there needs to be some way for this token to be refreshed, in case the apt update takes longer than the expiry time of the OAuth token, or in case the token in the environment is already expired or nearly expired.

Some ideas for refreshing the token:

1. Can the CI/CD system be trusted to update the env var with a fresh token when this one expires? If so, you could write your own refresher that pulls the new token from the environment when this one has expired.
2. If not, how does the CI/CD system get the token in the first place? Could that method be used to refresh (and possibly import to this plugin) a token?

Reply:
I am looking at potentially doing something similar in the yum plugin for artifact registry. So, before I embark on that, I want to contribute to this discussion as it seems most impactful to getting something like this merged.

1. I think the idea is that this is invoked in a build environment where there likely isn't an expectation for token refresh to be supported. The CI/CD system would expect the image build to fail should the time for build (inclusive of the time to invoke apt) exceeds the access token lifetime. 
2. The token could be issued with something as simple as using the print-access-token command to impersonate a service account. This would bring feature parity to how Artifact Registry has documented the behavior for usage with Docker here: https://cloud.google.com/artifact-registry/docs/docker/authentication#token

Reply:
> the idea is that this is invoked in a build environment where there likely isn't an expectation for token refresh to be supported.

Though this may work for your use case, changes to the plugin must work for all users. Not all users will assume the same context as you. As the person who will need to answer support requests when users come to us, saying "help, the plugin doesn't work," this isn't something we can feasibly do. There must be a refresh mechanism.

Reply:
@ericdand: How about alternatively supporting reading an access token from a file? This would follow the same principal as how the service account credential JSON file works. This would require the user to figure out a refresh mechanism to write a new access token to the file but likewise gives flexibility to users who won't use / don't want a refresh mechanism.

Reply:
If the user is willing to write a credential to disk (I assume that fetching the token from the environment is intended to avoid writing it to disk), then there's already `google.CredentialsFromJSON` that works even better than just writing a temporary token to a local file. ;)

> This would require the user to figure out a refresh mechanism to write a new access token to the file but likewise gives flexibility to users who won't use / don't want a refresh mechanism.

Re-fetching from _somewhere_ is a fine idea, though. I actually suggested something similar in my original comment, where I said: 

> Can the CI/CD system be trusted to update the env var with a fresh token when this one expires? If so, you could write your own refresher that pulls the new token from the environment when this one has expired.

I would consider it sufficient to add logic to this pull request that, if the token were found to be expired, would pull the token from the environment again, assuming that whatever wrote it to the environment would keep it fresh. This behaviour should be logged clearly (so that it's clear to the user that the plugin tried to get a new token out of the env var, but found that it was still expired), of course, but it would be sufficient as a refresh mechanism.

Reply:
@ericdand: I didn't see that CredentialsFromJSON works for access tokens. Can you provide an example of how that would work for an access token?

Reply:
Oh, it doesn't work for OAuth tokens -- but writing an OAuth token to disk is also not exactly something we recommend doing. ;) Use the usual JSON credentials if you'd like to use that method.

Reply:
@ericdand: The point is that we don't want long lived credentials. The idea is that we may have a builder process (Docker image, AMI building) that would use this to install packages and freeze the image. Having potentially long-lived JSON credentials isn't great.